### PR TITLE
Thomas/criteria

### PR DESCRIPTION
**Pull request by Mostah**

### CODE REVIEW

**Review by thibmonsel — APPROVED** [review]

**Review by thibmonsel — APPROVED** [review]

**Review by thibmonsel — APPROVED** [review]

File: src/Criteria.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#include "Criteria.h"
+#include "Criterion.h"
+#include <iostream>
+#include <string>
+#include <vector>
+
+Criteria::Criteria(std::vector<Criterion> &criterion_vect) {
+  // deep copy
+  for (int i = 0; i < criterion_vect.size(); i++) {
+    criterion_vect_.push_back(Criterion(criterion_vect[i]));
```
Comment (edix9) [inline src/Criteria.cpp:10]: why do we need to convert to Criterion criterion_vect[i] here?

File: test/TestCriteria.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
+#include "../src/Criteria.h"
+#include "../src/Criterion.h"
+#include "gtest/gtest.h"
+#include <sstream>
+#include <utility>
+
+TEST(TestCriteria, TestBaseConstructorWithNbAndId) {
+  std::string id = "test";
+  Criteria crits = Criteria(2, id);
```
Comment (edix9) [inline test/TestCriteria.cpp:9]: we should be careful here with variables name ("id" here is used for another purpose in the next lines)

**Review by edix9 — APPROVED** [review]
looks good

File: test/TestCriteria.cpp (hunk 2 above)
Comment (thibmonsel) [reply test/TestCriteria.cpp:9]: what do you mean ? 
id is just for the testing.

**Pull request merged**